### PR TITLE
WIP: update to 17.30

### PR DESCRIPTION
**Pull request by corngood**

### CODE REVIEW

**Comment by corngood** [comment]
This is meant to work with linux 4.10 and xorg 1.19.

I had to compile my kernel with `KALLSYMS_ALL y`.  Without it you will probably get an error like:

```
kernel: Error: fail to get symbol drm_gem_prime_dmabuf_ops
kernel: ------------[ cut here ]------------
kernel: kernel BUG at /tmp/nix-build-amdgpu-pro-17.30-4.9.47.drv-0/amdgpu-pro-17.30-465504/root/usr/src/amdgpu-pro-17.30-465504/amd/amdkcl/kcl_common.h:34!
```

check your booted kernel with:

```
$ gunzip -c /proc/config.gz | grep KALLSYMS_ALL
CONFIG_KALLSYMS_ALL=y
```

**Comment by ranisalt** [comment]
Both `linux` and `linux-zen` for Arch Linux are compiled with `KALLSYMS_ALL` unset :(

**Comment by corngood** [comment]
From https://packages.ubuntu.com/source/xenial/linux-hwe, it looks like they enable KALLSYMS_ALL in ubuntu 16.04.3.

As far as I can tell, the only other option (without major changes to the kernel module) would be to patch the kernel to make `drm_gem_prime_dmabuf_ops` public.  Then you're trading a kernel config for a patch, which doesn't feel like an improvement.

I would suggest trying `linux-lts` (4.9) modified with `KALLSYMS_ALL=y`, and then try updating it to 4.10 if it doesn't work.  I hit the `drm_gem_prime_dmabuf_ops` thing first on 4.9, and assumed it was incompatible, but it also happened on 4.10, and I never tried 4.9 again after fixing it.

**Comment by Tuerai** [comment]
I recompiled my kernel to have KALLSYMS_ALL=y, but I can't get all the dependencies to build this. Specifically, hsakmt-roct won't build for me.

File: .SRCINFO [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
+	depends = lib32-llvm-amdgpu-pro-5.0-dev=17.30.465504-2
+	depends = lib32-llvm-amdgpu-pro-5.0-runtime=17.30.465504-2
+	depends = lib32-llvm-amdgpu-pro-5.0=17.30.465504-2
+	depends = lib32-llvm-amdgpu-pro-runtime=17.30.465504-2
+	depends = lib32-ncurses5-compat-libs>=6
+	depends = lib32-zlib>=1.2.0
+	depends = libedit2>=2.11-20080614
+	depends = libffi-dev>=3.0.9
+	depends = libjs-jquery
+	depends = libjs-underscorea
```
Comment (azamat-sharapov) [inline .SRCINFO]: Typo? (libjs-underscore)

**Comment by Qiu3344** [comment]
I have a problem with hsakmt-roct too. I tried linux-lts and linux 4.10 and it's not compiling on both.

**Comment by cyborg00222** [comment]
@Qiu3344 https://github.com/RadeonOpenCompute/ROCT-Thunk-Interface/issues/14

**Comment by Qiu3344** [comment]
@cyborg00222 So how to fix it? Should I edit "CMakeList.txt"?

File: .SRCINFO (hunk 1 above)
Comment (corngood) [reply .SRCINFO]: Not a direct typo (these are generated from the .debs), but it does seem wrong.

**Comment by corngood** [comment]
What's the dependency chain leading to `hsakmt-roct`?

**Comment by Qiu3344** [comment]
"amdgpu-pro" requies "hsakmt-roct".

**Comment by corngood** [comment]
@Qiu3344 I don't see it in the .SRCINFO... are you sure it's not indirect?  In any case, if you need to fudge dependencies, you should be able to do it at:

https://github.com/corngood/archlinux-amdgpu/blob/5a5490f960d324d5cf3830a0e81514040eaf4524/gen-PKGBUILD.py#L230

**Comment by Qiu3344** [comment]
@corngood Thanks! This will probably fix the problem. Did you ever think about documenting the package?

**Comment by corngood** [comment]
@Qiu3344 Not seriously.  I inherited it from someone else, and it got an overhaul in the last update from another contributor.

If anyone's interested in taking over maintainership, let me know.

**Comment by Qiu3344** [comment]
@corngood How to apply changes that i made in gen-PKGBUILD.py?

**Comment by corngood** [comment]
@Qiu3344 Your best reference is the travis build script:

https://github.com/corngood/archlinux-amdgpu/blob/17.30/.travis.yml

At the top it lists the arch deps, then theres `sudo pip install python-debian`, and `./build.sh`.  That should be all you need to generate the PKGBUILD.

**Comment by Qiu3344** [comment]
@corngood when I run build.sh I get an error that amdgpu-pro is i386 and should begin with lib32.

**Comment by heavysink** [comment]
Successfully built, installed and run without problem with my RX550 on Linux 4.13.6 and Xorg 1.19.5 with some modifications. Three main modifications:
1. Ignore the DKMS module. 
2. Delete the libgbm.so* under /usr/lib from mesa-noglvnd. It is causing compatibility problem (undefined symbol) to the driver. I have made mesa-noglvnd-nogbm package on ArchlinuxCN repo.
3. Remove glamor related extract_deb inside PKGBUILD and use glamor provided by xorg-server package. It is also causing undefined symbol problem.

**Comment by brainpower** [comment]
@Qiu3344 : the i386 / lib32 error is fixed by the update to the packages_map in commit 5d02f39 of PR #54 .

**Comment by corngood** [comment]
Thanks @brainpower.  I'll close this one in favour of #54.  Anyone who can, please test and comment on it.

**Pull request closed without merging**